### PR TITLE
test(firestore): Resolve flaky tests

### PR DESCRIPTION
**Pull request by bhshkh**
Fixes: #10301, #10931, #10842, #11254

More info:

https://github.com/googleapis/google-cloud-go/issues/11254#issuecomment-2540273682

https://github.com/googleapis/google-cloud-go/issues/10301#issuecomment-2540243465

https://github.com/googleapis/google-cloud-go/issues/10931#issuecomment-2540223561

https://github.com/googleapis/google-cloud-go/issues/10842#issuecomment-2540199913

### CODE REVIEW

File: firestore/go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module cloud.google.com/go/firestore
 go 1.21
 
 require (
-	cloud.google.com/go v0.116.0
+	cloud.google.com/go v0.117.0
```
Comment (gkevinzheng) [inline firestore/go.mod:6]: Why is a test fix PR also updating the version requirement of a dependency for customers?

**Review by gkevinzheng — COMMENTED** [review]
The tests LGTM, just have one remark:

File: firestore/go.mod (hunk 1 above)
Comment (bhshkh) [reply firestore/go.mod:6]: to get Fatalf in testutil.Retry

**Review by gkevinzheng — APPROVED** [review]
LGTM

**Pull request merged**